test/jdk/java/awt/ScrollPane/ScrollPaneWindowsTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4452612
+  @requires os.family=="windows"
+  @summary The popup menu of the scroll bar doesn't work properly in Window2000.
+  @key headful
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Insets;
+import java.awt.Panel;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+import java.awt.ScrollPaneAdjustable;
+
+import java.awt.event.AdjustmentListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+public class ScrollPaneWindowsTest implements AdjustmentListener {
+    ScrollPane sp;
+    Panel p;
+    Robot robot;
+    Frame frame;
+    Insets paneInsets;
+    public static Object LOCK = new Object();
+    ScrollPaneAdjustable vScroll;
+    ScrollPaneAdjustable hScroll;
+    boolean notifyReceived = false;
+
+    public static void main(String[] args) throws Exception {
+        if (Double.valueOf(System.getProperty("os.version")) < 5.0) {
+            System.out.println("This test is for Windows 2000 and above.");
+            return;
+        }
```

Comment:
Shall this be dropped too? Java won't start on Windows versions before Vista (6.0), so does it serve a purpose other than a historical reasons?

Reply:
Updated.